src/ams.py

```diff
@@ -87,9 +87,25 @@ class PrometheusConfig:
 
     def __post_init__(self):
         """Post initialization validations."""
-        if self.port > 0:
+        if self.target_port > 0:
             self.enabled = True
 
+    @property
+    def scrape_jobs(self) -> List[Dict]:
+        """Generate scrape jobs for prometheus."""
+        job = {
+            "job_name": "metrics",
+            "metrics_path": self.metrics_path,
+            "static_configs": [{"targets": [f"{self.target_ip}:{self.target_port}"]}],
+        }
+        if self.basic_auth_password and self.basic_auth_password:
```

Comment:
if self.basic_auth_password and self.basic_auth_password:
->
 if self.basic_auth_username and self.basic_auth_password: